examples/reliable_conversation/test_basic.py

```
@@ -0,0 +1,340 @@
+            has_anthropic = hasattr(test_app.context.config, 'anthropic') and test_app.context.config.anthropic
+            
+            if not (has_openai or has_anthropic):
+                runner.formatter.show_warning("No LLM providers configured. Tests will use fallbacks.")
+                runner.formatter.console.print("   [dim]To test with real LLMs, add API keys to mcp_agent.secrets.yaml[/dim]")
+            else:
+                provider = "openai" if has_openai else "anthropic"
+                runner.formatter.show_success(f"LLM provider available: {provider}")
+
+            # Add filesystem access to current directory
+            if hasattr(test_app.context.config, 'mcp') and test_app.context.config.mcp:
+                if "filesystem" in test_app.context.config.mcp.servers:
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)